packages/arm-cluster/src/components/MarkerClusterGroup.tsx

```diff
@@ -76,21 +100,37 @@ const MarkerClusterGroup: React.FC<Props> = ({
     return null
   }, [mapInstance, optionsOverrides])
 
+  useEffect(() => {
+    if (setInstance && markerClusterGroup) {
```

Comment:
You can replace it with this:
```suggestion
    if (!setInstance || !markerClusterGroup) {
      return undefined
    }
```
So then you won't need to check this in the cleanup fn

Reply:
I think it's easier to reason about variables that do exist, it reads a bit better.

Reply:
It's a matter of personal preference, indeed. But if you use the statement like I suggested, you won't need to duplicate the statement in the cleanup function :) Therefore more clean and still readable

Reply:
Good argument. I've changed all the effects in this file to this approach.